var/spack/repos/builtin/packages/wrf-hydro/package.py

```diff
@@ -0,0 +1,231 @@
+        tty.msg("NetCDF-C installed to: " + str(nc_c_home))
+
+    def install(self, spec, prefix):
+        map_variant = lambda v: "1" if v else "0"
+        f90_flags = []
+
+        # Initialize variables used in WRF-Hydro compilation
+        wrf_envar = {
+            "WRF_HYDRO": "1",
+            "HYDRO_D": map_variant(self.spec.variants["hydro_d"].value),
+            "SPATIAL_SOIL": map_variant(self.spec.variants["spatial_soil"].value),
+            "WRF_HYDRO_RAPID": map_variant(self.spec.variants["wrf_hydro_rapid"].value),
+            "NCEP_WCOSS": map_variant(self.spec.variants["ncep_wcoss"].value),
+            "WRFIO_NCD_LARGE_FILE_SUPPORT": map_variant(
+                self.spec.variants["wrfio_ncd_large_file_support"].value
+            ),
+            "WRF_HYDRO_NUDGING": map_variant(
+                self.spec.variants["wrf_hydro_nudging"].value
+            ),
+        }
+
+        if self.spec.variants["debug"].value:
+            f90_flags += ["-g"]
+
+        # Set the compiler flag for ./configure based on active
+        # compiler family
+        compiler_config = {"pgi": "1", "gcc": "2", "intel": "3"}
+
+        try:
+            configure_args = [compiler_config[self.spec.compiler.name]]
+        except KeyError:
+            raise InstallError(
+                "Compiler not recognized nor supported: {}".format(
+                    self.spec.compiler.name
+                )
+            )
+
+        # Reconstruct the dict as a bash script
+        set_envar_sh = ["#!/bin/bash"] + [
+            "export %s=%s" % (key, wrf_envar[key]) for key in wrf_envar.keys()
+        ]
+
+        tty.msg("WRF-Hydro environment variables set to: ")
+        tty.msg("\n".join(set_envar_sh))
+
+        # Configure and build in trunk/NDHMS rather than root source
+        with working_dir(join_path("trunk", "NDHMS")):
+            copy(join_path("template", "setEnvar.sh"), "setEnvar.sh")
+            configure(*configure_args)
+            start_build = Executable("./compile_offline_NoahMP.sh")
+
+            # Write out Fortran flags, if the f90_flags list isn't empty...
+            if len(f90_flags):
+                with open("./macros", "a") as f_macros:
+                    f_macros.write("F90FLAGS+=%s\n" % (" ".join(f90_flags)))
+
+            with tempfile.NamedTemporaryFile(mode="w") as fp:
+                fp.write("\n".join(set_envar_sh))
+                fp.flush()
+                start_build(fp.name)
+
+        # Install compiled binaries
+        mkdir(prefix.bin)
+
+        install(join_path("trunk", "NDHMS", "Run", "wrf_hydro.exe"), prefix.bin)
+        install(
+            join_path("trunk", "NDHMS", "Run", "wrf_hydro_NoahMP.exe"),
+            prefix.bin,
+        )
+
+        # Install the entire tree as many files are needed for running
+        install_tree(".", prefix)
+
+    @run_after("install")
+    @on_package_attributes(run_tests=True)
+    def build_test(self):
```

Comment:
Note if you call this method "test" it will run when you `spack install --test=...`